From da46fd1690b4546e115abfd8a85f68996e415aeb Mon Sep 17 00:00:00 2001
From: trickypr <trickypr@icloud.com>
Date: Mon, 18 Jul 2022 13:40:19 +1000
Subject: [PATCH] Force build target to be firefox

This is used to allow for make watch, an incredebly helpful command
---
 python/mozbuild/mozbuild/base.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/python/mozbuild/mozbuild/base.py b/python/mozbuild/mozbuild/base.py
index e2a6feb59f01..6a3bb2fc8078 100644
--- a/python/mozbuild/mozbuild/base.py
+++ b/python/mozbuild/mozbuild/base.py
@@ -967,9 +967,9 @@ class MachCommandConditions(object):
     @staticmethod
     def is_firefox(cls):
         """Must have a Firefox build."""
-        if hasattr(cls, "substs"):
-            return cls.substs.get("MOZ_BUILD_APP") == "browser"
-        return False
+
+        # snippetman - force firefox for mach watch
+        return True
 
     @staticmethod
     def is_jsshell(cls):
@@ -1018,7 +1018,9 @@ class MachCommandConditions(object):
         """Must have a build."""
         return MachCommandConditions.is_firefox_or_android(
             cls
-        ) or MachCommandConditions.is_thunderbird(cls)
+        ) or MachCommandConditions.is_thunderbird(cls
+                                                  ) or True  # Quark - Force the run method to work
+        # TODO: Better solution for this
 
     @staticmethod
     def has_build_or_shell(cls):
-- 
2.37.1

